### DISCUSSION TITLE
show the subscriber users in each topic with adonisjs websocket

### DISCUSSION THREAD

**Discussion by farshadfahimi** [adonisjs/core#1691] [Help] [answered]
I have multiple topics and user can subscribe to them.

right now I need to show the list of each subscriber in own topic.
How should I do this?

for example:
we have two topic `X` & `Y`
I have 5 subscriber in `X` topic and `2` subscriber in `Y` topic.

I need to show the `X` subscriber in the `X` page and `Y` subscriber in `Y` page

**Comment by McSneaky** [comment, marked as answer]
Late to the party..
But you need to keep track of them manually. You can add users to some Set or Object when they connect and remove them from there when they disconnect. 
Then add some endpoint / WS event call to get the list of users.
Might wan't to keep them in DB or in Redis or somewhere for persistent store. I just added them to in-memory Object

**Reply by farshadfahimi** [reply]
@McSneaky 

I'm a little nuby in use nodejs and adonis. can you give me a sample code to know how should I store users in the memory as object?
